In [1]:
import csv

file = open("guns.csv") 
csvreader = csv.reader(file)
data = list(csvreader)

data[:5]

[['',
  'year',
  'month',
  'intent',
  'police',
  'sex',
  'age',
  'race',
  'hispanic',
  'place',
  'education'],
 ['1',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '34',
  'Asian/Pacific Islander',
  '100',
  'Home',
  '4'],
 ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'],
 ['3',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '60',
  'White',
  '100',
  'Other specified',
  '4'],
 ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]

In [2]:
headers = data[0] # Only if we want to remove header
data = data[1:] # Removes headers from data

headers

data[0:5]



[['1',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '34',
  'Asian/Pacific Islander',
  '100',
  'Home',
  '4'],
 ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'],
 ['3',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '60',
  'White',
  '100',
  'Other specified',
  '4'],
 ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'],
 ['5',
  '2012',
  '02',
  'Suicide',
  '0',
  'M',
  '31',
  'White',
  '100',
  'Other specified',
  '2']]

In [3]:
years = [row[1] for row in data]    

years

year_counts = {}

for item in years:
    if item in year_counts:
        year_counts[item] += 1
    else:
        year_counts[item] = 1
        
year_counts


{'2012': 33563, '2013': 33636, '2014': 33599}

In [49]:
import datetime

dates = [datetime.datetime(year = int(row[1]), month = int(row[2]), day = 1) for row in data]

print(dates[0:5])

date_counts = {}

for item in dates:
    if item in date_counts:
        date_counts[item] += 1
    else:
        date_counts[item] = 1
        
date_counts

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]


{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

In [4]:
sex_counts = {}
race_counts = {}
for item in data:
    sex = item[5]
    race = item[7]
    if sex in sex_counts:
        sex_counts[sex] += 1
    else:
        sex_counts[sex] = 1
    
    if race in race_counts:
        race_counts[race] += 1
    else:
        race_counts[race] = 1

race_counts

sex_counts
        


{'F': 14449, 'M': 86349}

# Gun Deaths In The US #
## From 538 Data, 2012 - 2014

So far we've done some exploratory data analysis of gun deaths in the US from the years 2012 - 2014. What' interesting to me so far is:

* The majority of gun deaths by Race are White (~66,237) and next is Black with (23,296), then Hispanic (9,022), then Asian/Pacific Islander (1,326) and finally Native American / Native Alaskan (917)
* The vast majority of gun deaths affect Men (~85% or 86349 / 100798)
* What needs further examination is type of death (i.e. Suicide, Homicide etc.) by sex and/or race and whether the police were involved in the death, by sex and race, both from the # of deaths and type.


In [5]:
census_file = open("census.csv", "r")
csvreader = csv.reader(census_file)
census = []

for item in csvreader:
    census.append(item)
    
census

[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

In [12]:
mapping = {
    "Asian/Pacific Islander": 15159516 + 674625,
    "Black": 40250635,
    "Native American/Native Alaskan": 3739506,
    "Hispanic": 44618105,
    "White": 197318956,
}

race_per_hundredk = {}

for rc_key, rc_value in race_counts.items():
    for m_key, m_value in mapping.items():
        if rc_key == m_key:
            race_per_hundredk[rc_key] = (rc_value / m_value) * 100000

print(race_per_hundredk)

{'Hispanic': 20.220491210910907, 'Native American/Native Alaskan': 24.521955573811088, 'Black': 57.8773477735196, 'Asian/Pacific Islander': 8.374309664161762, 'White': 33.56849303419181}


In [16]:
intents = [row[3] for row in data]

print(intents[0:5]) # To verify list_comprehension

homicide_race_counts = {}

for i, race in enumerate(races):
    if intents[i] == "Homicide":
        if race in homicide_race_counts:
            homicide_race_counts[race] += 1
        else:
            homicide_race_counts[race] = 1

homicide_race_counts

homicide_per_hundredk = {}

for hr_key, hr_value in homicide_race_counts.items():
            homicide_per_hundredk[hr_key] = (hr_value / mapping[hr_key]) * 100000

homicide_per_hundredk

['Suicide', 'Suicide', 'Suicide', 'Suicide', 'Suicide']


{'Asian/Pacific Islander': 3.530346230970155,
 'Black': 48.471284987180944,
 'Hispanic': 12.627161104219914,
 'Native American/Native Alaskan': 8.717729026240365,
 'White': 4.6356417981453335}

After only capturing gun deaths due to Homicide, we then mapped out gun deaths by race per 100,000 people, using the census data to accurate reflect gun deaths based on the proportion of people of that race in the US.

What the data shows is ***African Americans have an extremely high rate of homicidal gun deaths compared to other races***. African Americans have ~49 people per 100,000 die from homicidal gun deaths versus Whites at ~5 people per 100,000, Hispanics at ~13 per 100,000, Native American / Native Alaskan's at ~9 per 100,000 and Asian/Pacific Islanders at ~4 per 100,000.

For next steps I'd like to explore:
* Does the intent of death outside of homicide also vary by race?
* Do gun deaths it affect one gender of race more? (i.e. are African American men affected more than African American women by gun deaths?)